cmd/vsphere-csi/main.go

```diff
@@ -66,7 +70,37 @@ func main() {
 		log.Error("CSI endpoint cannot be empty. Please set the env variable.")
 		os.Exit(1)
 	}
+	log.Info("enable logging off for vCenter sessions on exit")
+	// Disconnect VC session on restart
+	defer func() {
+		if r := recover(); r != nil {
+			log.Info("cleaning up vc sessions")
+			cleanupSessions(ctx, r)
+		}
+	}()
+
+	ch := make(chan os.Signal, 1)
+	signal.Notify(ch, syscall.SIGTERM)
+	go func() {
+		for {
+			sig := <-ch
+			if sig == syscall.SIGTERM {
+				log.Info("SIGTERM signal received")
+				utils.LogoutAllvCenterSessions(ctx)
+				os.Exit(0)
+			}
+		}
+	}()
 
 	vSphereCSIDriver := service.NewDriver()
 	vSphereCSIDriver.Run(ctx, CSIEndpoint)
+
+}
+
+func cleanupSessions(ctx context.Context, r interface{}) {
+	log := logger.GetLogger(ctx)
+	log.Error("observed a panic and a restart was invoked, panic: ", r)
```

Comment:
This should be `log.Errorf("observed a panic and a restart was invoked, panic: %+v", r)` ?